gamehub-backend/src/test/java/com/example/gamehubbackend/controllers/ReviewControllerIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+                                "content": "Great game!",
+                                "date": "2020-01-01"
+                            }
+                        ]
+                        """
+                ));
+    }
+
+    @Test
+    @WithMockUser
+    void getReviewsByGameId_whenNotFound() throws Exception {
+        mockMvc.perform(get("/api/reviews/game1"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"));
+    }
+
+    @Test
+    @WithMockUser
+    @DirtiesContext
+    void addReview() throws Exception {
+        mockMvc.perform(post("/api/reviews")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(
+                                """
+                                {
+                                  "userId": "user1",
+                                  "gameId": "game1",
+                                  "username": "username1",
+                                  "rating": 4.5,
+                                  "content": "Great game!",
+                                  "date": "2020-01-01"
+                                }
+                                """
+                        ))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(get("/api/reviews/game1"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(
+                        """
+                        [
+                            {
+                                "userId": "user1",
+                                "gameId": "game1",
+                                "username": "username1",
+                                "rating": 4.5,
+                                "content": "Great game!",
+                                "date": "2020-01-01"
+                            }
+                        ]
+                        """
+                ))
+                .andExpect(jsonPath("$[0].id").exists());
+
+    }
+
+    @Test
+    @WithMockUser
+    @DirtiesContext
+    void updateReview() throws Exception {
+        reviewRepository.save(new Review("1", "user1", "game1", "username1", 4.5, "Great game!", "2020-01-01"));
+
+        mockMvc.perform(put("/api/reviews/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(
+                                """
+                                {
+                                  "userId": "user1",
+                                  "gameId": "game1",
+                                  "username": "username1",
+                                  "rating": 4.8,
+                                  "content": "Amazing game!",
+                                  "date": "2020-01-01"
+                                }
+                                """
+                        ))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(get("/api/reviews/game1"))
```

Comment:
if you change the update method, then you could just test the return value of the put request